x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/mapper/SemanticTextInferenceResultFieldMapper.java

```diff
@@ -0,0 +1,241 @@
+    }
+
+    private static void parseFieldInferenceResultsArray(
+        DocumentParserContext context,
+        MapperBuilderContext mapperBuilderContext,
+        String fieldName) throws IOException {
+
+        XContentParser parser = context.parser();
+        NestedObjectMapper nestedObjectMapper = createNestedObjectMapper(context, mapperBuilderContext, fieldName);
+        context.path().add(fieldName); // TODO: Need path manipulation?
+
+        try {
+            if (parser.nextToken() != XContentParser.Token.START_ARRAY) {
+                throw new DocumentParsingException(parser.getTokenLocation(), "Expected a START_ARRAY, got " + parser.currentToken());
+            }
+
+            for (XContentParser.Token token = parser.nextToken(); token != XContentParser.Token.END_ARRAY; token = parser.nextToken()) {
+                DocumentParserContext nestedContext = context
+                    .createChildContext(nestedObjectMapper)
+                    .createNestedContext(nestedObjectMapper);
+
+                if (token != XContentParser.Token.START_OBJECT) {
+                    throw new DocumentParsingException(parser.getTokenLocation(), "Expected a START_OBJECT, got " + parser.currentToken());
+                }
+
+                for (token = parser.nextToken(); token != XContentParser.Token.END_OBJECT; token = parser.nextToken()) {
+                    if (token != XContentParser.Token.FIELD_NAME) {
+                        throw new DocumentParsingException(parser.getTokenLocation(),
+                            "Expected a FIELD_NAME, got " + parser.currentToken());
+                    }
+
+                    String currentName = parser.currentName();
+                    context.path().add(currentName);
+                    try {
+                        // TODO: Test how this code handles extra/missing fields
+                        FieldMapper childNestedMapper = (FieldMapper) nestedObjectMapper.getMapper(currentName);
+                        if (childNestedMapper == null) {
+                            throw new DocumentParsingException(parser.getTokenLocation(), "Unexpected field name: " + currentName);
+                        }
+                        parser.nextToken();
+                        childNestedMapper.parse(nestedContext);
+                    } finally {
+                        context.path().remove();
+                    }
+                }
+            }
+        } finally {
+            context.path().remove();
+        }
+    }
+
+    private static NestedObjectMapper createNestedObjectMapper(
+        DocumentParserContext context,
+        MapperBuilderContext mapperBuilderContext,
+        String fieldName) {
+
+        // TODO: Use keyword field type for text?
+        // TODO: Why add text field to mapper if it's not indexed or stored?
+        SparseVectorFieldMapper.Builder sparseVectorMapperBuilder = new SparseVectorFieldMapper.Builder(SPARSE_VECTOR_SUBFIELD_NAME);
```

Comment:
I changed the name of the inference sub-field to `sparse_embedding` to make it explicitly clear that it's a sparse vector. That way, when we add support for dense vectors, we can add a `dense_embedding` sub-field and there won't be any ambiguity around inference sub-field types. Thoughts?

Reply:
That can potentially help on distinguishing both cases. I'm not sure right now about it - I'd say let's keep it under the same name and re-evaluate when we introduce dense vectors to avoid introducing a one-way door early on, but not a strongly held argument